src/components/UserSelector.tsx

```diff
@@ -1,40 +1,97 @@
-import React from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+type Props = {
+  users: User[];
+  setUserSelected: (user: User) => void;
+  userSelected: User | null;
+  handleUserPosts: (user: User) => void;
+  setShowSideBar: (item: number | null) => void;
+};
```

Comment:
You have not described objects in propTypes. According to the checklist, make sure you describe objects in propTypes. This is a code quality issue and it helps in validating the props passed to the component.